src/commands/compose/compose.ts

```diff
@@ -90,9 +96,12 @@ export async function composeRestart(context: IActionContext, dockerComposeFileU
 
 const serviceListPlaceholder = /\${serviceList}/i;
 const profileListPlaceholder = /\${profileList}/i;
-async function addServicesOrProfilesIfNeeded(context: IActionContext, workspaceFolder: vscode.WorkspaceFolder, command: string): Promise<string> {
+async function addServicesOrProfilesIfNeeded(context: IActionContext, workspaceFolder: vscode.WorkspaceFolder, command: string, preselectedServices: string[]): Promise<string> {
     const commandWithoutPlaceholders = command.replace(serviceListPlaceholder, '').replace(profileListPlaceholder, '');
-    if (serviceListPlaceholder.test(command) && profileListPlaceholder.test(command)) {
+
+    if (preselectedServices?.length && serviceListPlaceholder.test(command)) {
```

Comment:
This logic isn't quite right--the `${profileList}` placeholder isn't being replaced.